source/Server/IntegratedAuthenticationModule.cs

```diff
@@ -18,26 +22,34 @@ public IntegratedAuthenticationModule(ILog log, IAuthCookieCreator tokenIssuer,
                     return responseCreator.Unauthorized(Request);
 
                 var principal = (IOctopusPrincipal)Context.CurrentUser;
-                var authCookies = tokenIssuer.CreateAuthCookies(Context.Request, principal.IdentificationToken, SessionExpiry.TwentyDays);
 
-                var whitelist = authenticationConfigurationStore.GetTrustedRedirectUrls();
-                Response response;
-                if (Request.Query["redirectTo"].HasValue && Requests.IsLocalUrl(Request.Query["redirectTo"].Value, whitelist))
+                if (Request.Query["state"].HasValue)
                 {
-                    var redirectLocation = Request.Query["redirectTo"].Value;
-                    response = new RedirectResponse(redirectLocation).WithCookies(authCookies);
-                }
-                else
-                {
-                    if (Request.Query["redirectTo"].HasValue)
+                    var stateData = HttpUtility.UrlDecode(Request.Query["state"].Value);
+                    var state = JsonConvert.DeserializeObject<LoginState>(stateData);
+
+                    var authCookies = tokenIssuer.CreateAuthCookies(Context.Request, principal.IdentificationToken, SessionExpiry.TwentyDays, (bool)state.UsingSecureConnection);
```

Comment:
For some intangible reason I feel more comfortable seeing `state.UsingSecureConnection == true` rather than `(bool)state.UsingSecureConnection`.